client.go

```diff
@@ -3,64 +3,70 @@ package supabase
 import (
 	"errors"
 
+	"github.com/supabase-community/gotrue-go"
+	"github.com/supabase-community/gotrue-go/types"
+	postgrest "github.com/supabase-community/postgrest-go"
 	storage_go "github.com/supabase-community/storage-go"
-	"github.com/supabase/postgrest-go"
 )
 
 const (
 	REST_URL     = "/rest/v1"
 	STORGAGE_URL = "/storage/v1"
+	AUTH_URL     = "/auth/v1"
 )
 
 type Client struct {
-	rest    *postgrest.Client
-	Storage *storage_go.Client
+	// Why is this a private field??
+	rest    postgrest.Client
+	Storage storage_go.Client
+	Auth    gotrue.Client
+	options clientOptions
 }
 
-type RestOptions struct {
-	Schema string
-}
-
-type ClientOptions struct {
-	Headers map[string]string
-	Db      *RestOptions
+type clientOptions struct {
+	url     string
+	headers map[string]string
 }
 
 // NewClient creates a new Supabase client.
 // url is the Supabase URL.
 // key is the Supabase API key.
 // options is the Supabase client options.
-func NewClient(url, key string, options *ClientOptions) (*Client, error) {
+func NewClient(url, key string, schema string, headers map[string]string) (*Client, error) {
```

Comment:
pass schema and headers as arguments vs putting in a struct.

Reply:
I think we should heep it in struct will be better, in the feature, if we have more fields, we can put it in struct instead of  parameters. 
The second issue, I want to make it consistent format with `supabase-js` library:
```
const supabase = createClient('https://xyzcompany.supabase.co', 'public-anon-key', {
  global: {
    fetch: (...args) => fetch(...args),
  },
})
```